# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MIROps.yaml
# Commit: 28feda14fdec
# Full Hash: 28feda14fdec962ea68c26a0570ea69042a5ad27
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 3
# Description:
#   Bug 1692065 - wasm: Add tagging for JSString* in AnyRef. r=yury
#   
#   Before this commit we would need to box a JSString* in WasmValueBox
#   when we received one from JS and converted to an AnyRef.
#   
# ==============================================================================

diff -r 85605041672d -r 28feda14fdec js/src/jit/MIROps.yaml
--- a/js/src/jit/MIROps.yaml	Fri Aug 04 14:06:50 2023 +0000
+++ b/js/src/jit/MIROps.yaml	Fri Aug 04 14:06:50 2023 +0000
@@ -570,9 +570,7 @@
 - name: WasmTruncateToInt32
   gen_boilerplate: false
 
-# Store a JS Value that can't be represented as an AnyRef pointer into an
-# object that holds the value (opaquely) as such a pointer.
-- name: WasmBoxValue
+- name: WasmAnyRefFromJSValue
   operands:
     def: Value
   result_type: WasmAnyRef
@@ -587,6 +585,14 @@
   congruent_to: if_operands_equal
   alias_set: none
 
+- name: WasmAnyRefFromJSString
+  operands:
+    def: String
+  type_policy: none
+  result_type: WasmAnyRef
+  congruent_to: if_operands_equal
+  alias_set: none
+
 - name: Int32ToIntPtr
   gen_boilerplate: false
 
@@ -1028,10 +1034,10 @@
 - name: WasmTrap
   gen_boilerplate: false
 
-# Trap if the given value is null
+# Trap if the given ref is null
 - name: WasmTrapIfNull
   operands:
-    value: RefOrNull
+    ref: WasmAnyRef
   arguments:
     trap: wasm::Trap
     bytecodeOffset: wasm::BytecodeOffset